src/lib/link.ts

```diff
@@ -258,6 +264,9 @@ export class Link {
     source: Side,
     minHeight: number
   ): Promise<number> {
+    this.logger.info(
+      `Update client for source ${source} to min height ${minHeight}`
```

Comment:
This will call the above method (with log) if we need to update.
Maybe, `this.logger.info(`Checking if client for source ${source} >= height ${minHeight}` or something similar.